fix: make search experience smoother

xynydev opened the pull request:
- load search results serverside
- add a loading bar to the top
- stop loading search completions after user has searched

ColinEspinas commented:
Thanks for the contribution, 
I find this change interesting as this would also kind of fix #20 by just waiting for the user to submit.
I'll come back to this once I have more feedback on the whole "wait for submit vs realtime updates" situation.

> To test this I was serving the app locally using python's http.server module. For this to work I had to move the assets and the config.js into the src directory. Is there a better way to run the app locally that doesn't require those changes?

I just use the liveserver extension on VSCode, it works by serving the root of the repository and just navigating to the `/src` route. I would agree that this is not a great setup. We are already looking into changing the technologies we use for the front-end and the extension. As this pure vanilla front-end is great for a prototype but not ideal for contributors to engage with.

I will probably talk about it more on element, join us if you've got some time to chat.

ColinEspinas commented on src/components/organisms/search-bar.js:
```diff
@@ -1,37 +1,40 @@
 
 const template = () => /*html*/`
-  <div class="search-bar">
+  <form class="search-bar" method="GET" action="/">
```
I don't think we need the `method` and `action` attributes here. Did you include them for any specific reason?

sarayourfriend commented:
> just navigating to the /src route

Oh dear, that's quite obvious now :facepalm: I was serving the `src` directory specifically instead of the root of the repo :sweat_smile: 

For the attributes: I think they are not necessary and wouldn't ever be used in a context where this page actually loads on the browser, given the rendering depends entirely on client side JS. If it was an SSR'd page then the form would make the input work even if JavaScript wasn't enabled. Given that's not the case I'll remove those attributes.

sarayourfriend commented:
We discussed this briefly in [#mwmbl:matrix.org chat](https://matrix.to/#/!MDsQmCFnIphgrxQipD:matrix.org/$8PaKxBPIhYoJIAleNlnrM6uK5OJjbbADyWv9m-0lmkI?via=matrix.org&via=matrix.tech-bloc-sea.dev&via=wiredentrypoint.xyz) and decided that an alternative approach to try would be to only wait for submit when reduced motion is enabled but also increase the debounce timeout for "real time search" higher than 100ms. I'll work on implementing that later today.

sarayourfriend commented:
@ColinEspinas changes are up, at your leisure :grin:

ColinEspinas commented on src/components/organisms/search-bar.js:
```diff
@@ -1,6 +1,9 @@
 import { globalBus } from '../../utils/events.js';
+import debounce from '../../utils/debounce.js'
+
+const prefersReducedMotion = window.matchMedia('(prefers-reduced-motion)').matches;
```
If we just use this once maybe just putting this condition into the if statement would be better. Do you think putting this outside will serve any purpose?

ColinEspinas replied:
Fine by me, will merge.

xynydev committed:
fix: make search experience smoother

- load search results serverside
- add a loading bar to the top
- stop loading search completions after user has searched
